### ISSUE TITLE
Mortar-mounted pawns with a Shieldbelt can't aim at the ground and keep the gun aiming without shooting

### ISSUE THREAD

**Issue by Alicecomma** [NoImageAvailable/CombatExtended#761]
- Prevent pawns with a shieldbelt to mount mortars;

OR

- Allow pawns with a shieldbelt to shoot when mounted.

**Comment by Alicecomma** [comment]
Original bug report on Steam (https://steamcommunity.com/profiles/76561198238877256):

CynSaje 14 Aug @ 2:02am 
Hi, I don't know if it is intended, but I discovered that if you have a shield belt while manning a mortar, the mortars go in a infinite reloading loop.

I don't know if this problem is present in vanilla though, but I never had it before installing this mod.

**Comment by NoImageAvailable** [comment]
Shield belts preventing turret operation is a vanilla feature.

**Comment by Alicecomma** [comment]
This isn't true:
![image](https://user-images.githubusercontent.com/4662406/63693512-8fdc6e80-c814-11e9-9233-eadd69a6b5f2.png)

**Comment by Alicecomma** [comment]
Same when drafted with the shield visibly active:
![image](https://user-images.githubusercontent.com/4662406/63693778-37f23780-c815-11e9-9241-13b61850d883.png)

**Comment by Alicecomma** [comment]
So the problem with the Harmony postfix currently is that it's logically impossible to form the expression (!A && !C) || B which is needed to validate the operation being allowed:

```

            /*
             *  A   B   C   D       E
             *  0   0   0   1       1
             *  0   0   1   1       0
             *  0   1   0   1       1
             *  0   1   1   1       1
             *  1   0   0   0       0
             *  1   0   1   0       0
             *  1   1   0   1       1
             *  1   1   1   1       1
             * 
             * A = verb is Verb_LaunchProjectile
             * B = ReachabilityImmediate(..)
             * C = verb is Verb_LaunchProjectile_CE
             * D = (!A) || B
             * E = (!A && !C) || B
             * 
             * We know A, C and D. Can we get E?
             * 
             *  A   C   D       F
             *  0   0   1       1
             *  0   1   1       0
             *  0   0   1       1
             *  0   1   1       1   ==> Duplicate of line 2, but a different result
             *  1   0   0       0
             *  1   1   0       0
             *  1   0   1       1
             *  1   1   1       1
             *  
             *  So, we can't get E from A, C and D alone. We need B.
             */
```

This means we should know the result of ReachabilityImmediate(..) or we can't get an accurate AllowVerbCast result accounting for Verb_LaunchProjectileCE.

**Comment by Alicecomma** [comment]
Interestingly, it can be solved by setting __result = ReachabilityImmediate(..) whenever A=0, C=1, D=1. This means that stuff is calculated twice, _but it shouldn't be such a resource hog_....

**Comment by Alicecomma** [comment]
With all that stuff fixed, the mortars still can't be fired unless they are fired at the pawn itself.

It seems the vanilla behaviour is somehow off or something? Or perhaps the CasterIsPawn check in CE actually returning the turret operator means it prevents the pawn from operating the turret - while in vanilla CasterIsPawn is false for manned turrets?

This seems to be the case:
- Manned turrets' "CasterPawn" is null, because the caster is the turretGun, not the operator.
- CE overrides this by checking ShooterPawn => CasterPawn ?? _turret operator_.
- Therefore, the CE manned turrets can't be shot by pawns with shieldbelts, because the caster is a pawn with a shieldbelt instead of a turretgun.

**Comment by Alicecomma** [comment]
```
[11:45 PM] LockdownX7: Regarding shield belts and mannable turrets, two things to consider:

- Direct-fire turrets can't allow shielded operators at all: suppression immunity and high health pool makes things ridiculous for the raiders, and even packs of HCB centis get mostly neutered by a KPV using rotating shielded operators.

- Mortars, on the other hand, are normally fired without direct line of sight, so at first glance, it shouldn't really matter if colonists can fire them while wearing shield belts. However, raiders can also use mortars, and after a quick test, it looks like shield-belt raiders do try to operate them during a siege (which they can't actually fire in the current version, so right now they just waste that mortar if it happens). If this is fixed so shield users can fire them, players will have to contend with mortar operators they can't feasibly snipe, which isn't ideal.

So with that, it doesn't seem to me like allowing shields on turret operators would be a positive change. However, one way I can think of to address both these problems while still maintaining balance and adding convenience, is to disable the shield effect while the pawn is operating a turret. It'd be functionally the same as dropping the belt on the floor before manning the turret, but without the busywork.
```

**Comment by Alicecomma** [comment]
Attempt for fix in #767 by Lockdown
- Ignore shieldbelt's AllowVerb check on manned turrets
- Set shieldbelt charge to 0 while manning
